src/common/getType.ts

```diff
@@ -545,22 +591,28 @@ function getOpCodeType(
       return voidType;
     case "text_replace": {
       expectType(textType(), textType(), textType());
-      const a = types[0] as TextType;
-      const c = types[2] as TextType;
-      return textType(a.capacity * c.capacity);
+      const [a, c] = [types[0], types[2]] as TextType[];
+      return textType(
+        getArithmeticType("mul", a.codepointLength, c.codepointLength),
```

Comment:
Could be narrowed by dividing by `b.codepointLength`, which would also check that b's length is not zero.

Reply:
> Could be narrowed by dividing by `b.codepointLength`, which would also check that b's length is not zero.

`text_replace "xyz" "abc" "X";` is `"xyz":(Text 3..3)` but dividing would suggest `(Text 1..1)`. Unless I missunderstood what you meant?

Reply:
You're right